okhttp-tests/src/test/java/com/squareup/okhttp/HttpUrlTest.java

```diff
@@ -1005,8 +1012,8 @@
   @Test public void composeQueryWithEncodedComponents() throws Exception {
     HttpUrl base = HttpUrl.parse("http://host/");
     HttpUrl url = base.newBuilder().addEncodedQueryParameter("a+=& b", "c+=& d").build();
-    assertEquals("http://host/?a%20%3D%26%20b=c%20%3D%26%20d", url.toString());
-    assertEquals("c =& d", url.queryParameter("a =& b"));
+    assertEquals("http://host/?a+%3D%26%20b=c+%3D%26%20d", url.toString());
+    assertEquals("c+=& d", url.queryParameter("a+=& b"));
```

Comment:
queryParameter takes the _unencoded_ value, and you're querying with the encoded value. This is incorrect.